index.js

```diff
@@ -2,6 +2,30 @@ import { tabbable, isFocusable } from 'tabbable';
 
 let activeFocusDelay;
 
+const delayFocusTrapActivation = (container, checkCanActivate, callback) => {
+  const startDate = Date.now();
+  const interval = setInterval(function () {
+    if (checkCanActivate(container)) {
+      clearInterval(interval);
+      callback();
+    } else {
+      const timeDifferenceInSeconds = (Date.now() - startDate) / 1000;
+      if (timeDifferenceInSeconds > 10) {
+        // eslint-disable-next-line no-console
+        console.warn(
+          [
+            'Focus-Trap activation for the following element timed out after',
+            timeDifferenceInSeconds,
+            'seconds',
+          ].join(' '),
```

Comment:
Actually, I'm thinking a warning is not necessary here. If we were producing Dev and Prod builds, then I'd say put this type of warning in the Dev build, but we only produce Prod builds, and they typically never log to the console, so it's nice, but we should leave this out.

I'm also wondering about the interval. It feels a bit aggressive, and tailored to a use case where there's a really short delay, and they how often should you check. That's really all subjective to the consumer's scenario. For you, it's like 500ms I think, but it could be more or less for someone else.

Is there a way we could change `checkCanActivate()` such that the caller can dictate when the next check occurs, and we use `setTimeout()` instead? The method returns a boolean, but maybe it could return the delay as a number in milliseconds before the check again? If it returns anything that is "not a number or `isNaN()`", then consider it "done" and proceed with activation. If you don't specify the option, then it's `undefined`, which is the equivalent of

```javascript
const result = checkCanActivate(containers);
if (typeof result !== 'number' && !isNaN(result)) {
  finishActivation()
} else {
  setTimeout(result, delayActivation.bind(null, containers, checkCanActivate, callback));
}
```
}

What do you think?

Reply:
Fair point about the customizable timing, I can see developers wanting to customize how frequently the checks are made.

If we remove this warning completely then how will developers know that they have an infinite loop bug and what is causing it? Suddenly performance on their site takes a massive hit and they will have no idea why.

Prod builds can have errors and warnings in the console. Website users don't see them and I'm certain that developers would appreciate knowing exactly why their site's performance has suddenly taken a nose dive if they write a bug in their code that causes an infinite loop.

Reply:
> For you, it's like 500ms I think, but it could be more or less for someone else.

It's actually only a few milliseconds for me. 

Yes the full animation takes 500ms but I only need to delay the activation long enough for `visibility` to transition from `hidden` to `visible` which is only a few milliseconds.